### ISSUE TITLE
🐛 noUnusedImports does not detect used types

### ISSUE THREAD

**Issue by EvHaus** [biomejs/biome#557]
### Environment information

```block
Incompatible Biome Server: ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

ℹ Rage discovered this running server using an incompatible version of Biome.

Server:
  Version:                      1.2.2
```

### What happened?

When the `nursery/noUnusedImports` rule is used in a file where a type is imported, it will detect that import as unused (even though it's actually used).

```js
import type NextAuth from 'next-auth/next';
const a: typeof NextAuth = {}
```

### Expected result

Because the import is actually used, it should not be flagged for removal.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
This is intentional. `typeof` can only be applied on a value, not a type. Are you sure that your code is correct? How is defined `NextAuth`?

**Comment by Louis-Michalak** [comment]
@Conaclos typeof can be applied to a type in that case

**Comment by Louis-Michalak** [comment]
I have the same issue on my side
```typescript
import type * as Config from './objects/config'; // -> Config is reported as unused
type ConfigCheck<T extends CheckObject<Omit<typeof Config, never>> = Omit<typeof Config, never>> =
  T;
```

**Comment by Conaclos** [comment]
In my view, `COnfig` should be imported as a value:

```ts
import * as Config from './objects/config'; // -> Config is reported as unused
type ConfigCheck<T extends CheckObject<Omit<typeof Config, never>> = Omit<typeof Config, never>> =
  T;
```

Taking some time to investigate, it seems that TypeScript allows the use of a type in `typeof` in only one case: if the type is imported. This could avoid conflicts with `verbatimModuleSyntax` mode. I have to investigate more to be sure about that.

**Comment by EvHaus** [comment]
Maybe we need a rule for detecting when you're unnecessarily using `typeof` on an imported type :)

**Comment by Conaclos** [comment]
Yes this conflicts with `verbatimModuleSyntax`.
I shipped a fix on main which allows using imported type in `typeof` position.
You can test on the playground that the fix solves your issue :)
